.ci/Jenkinsfile

```diff
@@ -314,43 +315,105 @@ def generateFunctionalTestStep(Map args = [:]){
 
   def goArch = "amd64"
```

Comment:
We may need to change goArch here for Windows workers 🤔

Reply:
I'm taking a look at https://golang.org/doc/install/source#environment, and it seems we do not need to change it, even for MacOS:
```
darwin   arm64
darwin   amd64
windows  386
windows  amd64
```

Reply:
Yea the architecture identifier should be similar across platforms. In the Linux world there was some differences between aarch64/arm64 but should hopefully be normalized within go internals as well